torch/csrc/distributed/c10d/NCCLUtils.hpp

```diff
@@ -300,8 +300,7 @@ class NCCLComm {
   }
 #endif
 
-#ifdef IS_NCCL_EXP
-#ifdef NCCL_COMM_DUMP
+#if defined(IS_NCCL_EXP) && defined(NCCL_COMM_DUMP)
```

Comment:
Thanks for the quick fix @shuqiangzhang ! I was wondering, maybe we don't have to check `IS_NCCL_EXP`, but `NCCL_COMM_DUMP` is enough?

Reply:
That was my first version implementation, but just make it double safe, Will suggested we check both, e.g., in unlikely cases, NCCL upstream might use a same macro but different implementation